### DISCUSSION TITLE
How can I tell MIP to not use GUROBI?

### DISCUSSION THREAD

**Discussion by erelsgl** [coin-or/python-mip#313] [Q&A]
I have GUROBI installed, and MIP automatically detects it and uses it. How can I tell MIP to not use GUROBI?

**Comment by sebheger** [comment]
``` class Model(name='', sense='MIN', solver_name='', solver=None) ```

When start building the model you can set solver_name to "cbc".

**Reply by rschwarz** [reply]
Would it be an idea to change the default behavior to using CBC, as it should always work?

Even if Gurobi is not installed (or no license is found) the default code path tries Gurobi first, which takes some time. When done repeatedly (eg in automated tests of tiny models) this can be significant.

**Comment by erelsgl** [comment]
I use the following code:

    try:
        model = mip.Model(name='', sense='MIN', solver_name='', solver=None)
    except mip.InterfacingError:
        model = mip.Model(name='', sense='MIN', solver_name='cbc', solver=None)

so that, if GUROBI is installed and its license is OK, it will run, otherwise MIP will use CBC.

It works, but I also get a lot of error messages:

```
Exception ignored in: <function SolverGurobi.__del__ at 0x7f343cc4f9d0>
Traceback (most recent call last):
  File "/mnt/d/Dropbox/ariel-algorithms/fairpy/.venv38linux/lib/python3.8/site-packages/mip/gurobi.py", line 439, in __del__
    GRBfreeenv(self._env)
TypeError: initializer for ctype 'GRBenv *' must be a pointer to same type, not cdata 'GRBenv * *'
```

Is there a way to suppress these error messages?

**Reply by sebheger** [reply]
I moved your error description to issue #325
